en/lessons/specifics/benchmarking.md

```diff
@@ -0,0 +1,346 @@
+  "map.flatten" => fn(list) -> list |> Enum.map(map_fun) |> List.flatten end
+}, inputs: inputs)
+```
+
+You'll notice two differences. First, we now have an `inputs` map that contains the information for our inputs to our functions. We're passing that inputs map as a configuration option to `Benchee.run/2`.
+
+And since our functions need to take an argument now, we need to update our benchmark functions to accept an argument, so instead of:
+```elixir
+fn -> Enum.flat_map(list, map_fun) end
+```
+
+we now have:
+```elixir
+fn(list) -> Enum.flat_map(list, map_fun) end
+```
+
+Let's run this again using:
+
+```shell
+$ mix run benchmark.exs
+```
+
+Now you should see output in your console like this:
+
+```shell
+Operating System: macOS
+CPU Information: Intel(R) Core(TM) i5-4260U CPU @ 1.40GHz
+Number of Available Cores: 4
+Available memory: 8.589934592 GB
+Elixir 1.5.1
+Erlang 20.0
+Benchmark suite executing with the following configuration:
+warmup: 2.00 s
+time: 5.00 s
+parallel: 1
+inputs: large list, medium list, small list
+Estimated total run time: 2.10 min
+
+
+
+Benchmarking with input large list:
+Benchmarking flat_map...
+
+Benchmarking with input medium list:
+Benchmarking flat_map...
+
+Benchmarking with input small list:
+Benchmarking flat_map...
+
+Benchmarking with input large list:
+Benchmarking map.flatten...
+
+Benchmarking with input medium list:
+Benchmarking map.flatten...
+
+Benchmarking with input small list:
+Benchmarking map.flatten...
+
+##### With input large list #####
+Name                  ips        average  deviation         median
+flat_map             6.29      158.93 ms    ±19.87%      160.19 ms
+map.flatten          4.80      208.20 ms    ±23.89%      200.11 ms
+
+Comparison:
+flat_map             6.29
+map.flatten          4.80 - 1.31x slower
+
+##### With input medium list #####
+Name                  ips        average  deviation         median
+flat_map           1.34 K        0.75 ms    ±28.14%        0.65 ms
+map.flatten        0.87 K        1.15 ms    ±57.91%        1.04 ms
+
+Comparison:
+flat_map           1.34 K
+map.flatten        0.87 K - 1.55x slower
+
+##### With input small list #####
+Name                  ips        average  deviation         median
+flat_map         122.71 K        8.15 μs   ±378.78%        7.00 μs
+map.flatten       86.39 K       11.58 μs   ±680.56%       10.00 μs
+
+Comparison:
+flat_map         122.71 K
+map.flatten       86.39 K - 1.42x slower
+```
+
+We can now see information for our benchmarks, grouped by input. This simple
+example doesn't provide any mind blowing insights, but you'd be very surprised
+how much performance varies based on input size!
+
+# Formatters
+
+The console output that we've seen is a helpful beginning for measuring the
+runtimes of your functions, but it's not your only option! In this section we'll
+look briefly at the three other available formatters, and also touch on what
+you'd need to do to write your own formatter if you like.
```

Comment:
Wow that's pretty in depth! :tada: Personally I'd say being able to handle one formatter would be enough :)